aten/src/ATen/native/cuda/ForeachUnaryOp.cu

```diff
@@ -221,6 +224,9 @@ std::vector<Tensor> foreach_tensor_neg_cuda(TensorList tensors) {
 
 void foreach_tensor_neg_cuda_(TensorList tensors) {
     check_foreach_api_restrictions(tensors);
+    TORCH_CHECK(tensors[0].scalar_type() != kBool,
+                "Negation, the `-` operator, on a bool tensor is not supported. "
+                "If you are trying to invert a mask, use the `~` or `logical_not()` operator instead.");
```

Comment:
Does `~` or `logical_not` work on TensorList right now?

Reply:
Should we improve the error message? Some things I am noticing:
1. If the user gets this error message, they did not actually use the `-` builtin operator. E.g. a user didn't do `tensors = [...]; other_tensors = -tensors`. So the "the `-` operator" part of the error message is confusing
2. The error message assumes that the user applied operations to a bool tensor. Maybe we should mention that the API being called is "_foreach_neg"
3. The error message shows up after a user calls `_foreach_neg`. But the error message suggests workarounds for a single tensor. This may be misunderstood to "just use ~ on your tensorlist"

Adding all the top together, maybe something like the following would make more sense?
```
_foreach_neg: There is a bool tensor in the passed-in TensorList.
Negation on a bool tensor is not supported. If you are trying to invert a mask, please
use the `~` or `logical_not()` operator on the individual tensors instead
```